app/src/org/commcare/services/WiFiDirectBroadcastReceiver.java

```diff
@@ -34,6 +38,9 @@ public WiFiDirectBroadcastReceiver(WifiP2pManager manager,
         super();
         this.manager = manager;
         this.activity = activity;
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.O_MR1) {
+            connectivityManager = (ConnectivityManager)activity.getContext().getSystemService(Context.CONNECTIVITY_SERVICE);
```

Comment:
Is there a specific reason we are initializing `connectivityManager` in constructor instead of when we need to use it later in `onReceive`.

Reply:
no reason, i'll move it